src/dispatch.rs

```diff
@@ -388,12 +388,25 @@ impl<'t> DispatcherBuilder<'t> {
     /// # Panics
     ///
     /// * if the specified dependency does not exist
-    pub fn add<T>(mut self, system: T, name: &str, dep: &[&str]) -> Self
+    pub fn add<T>(self, system: T, name: &str, dep: &[&str]) -> Self
```

Comment:
I think that it needs to be documented that the default id is `0`, or make it something like `usize::MAX`